foreman/data_refinery_foreman/surveyor/array_express_surveyor.py

```diff
@@ -0,0 +1,65 @@
+import requests
+from typing import List
+from data_refinery_models.models import Batch, BatchKeyValue, SurveyJob
+from .external_source import ExternalSourceSurveyor, ProcessorPipeline
+
+
+class ArrayExpressSurveyor(ExternalSourceSurveyor):
+    # Files API endpoint for ArrayExpress
+    FILES_URL = "http://www.ebi.ac.uk/arrayexpress/json/v2/files"
+
+    def source_type(self):
+        return "ARRAY_EXPRESS"
+
+    def determine_pipeline(self,
+                           batch: Batch,
+                           key_values: List[BatchKeyValue] = []):
+        return ProcessorPipeline.MICRO_ARRAY_TO_PCL
+
+    def survey(self, survey_job: SurveyJob):
+        # Obviously this shouldn't be hardcoded in but the Array Express
```

Comment:
No actually I think it should be a field on the SurveyJob. However it probably won't be an entity for each source we'll have so I think it should probably be in a key value table. I had considered making a key value table for SurveyJobs like I have for the Batches table but I guess I decided not too. This seems like a good reason to change my mind.